scripts/eslint.py

```diff
@@ -0,0 +1,62 @@
+"""  # lint-amnesty, pylint: disable=django-not-configured
+Check code quality using pycodestyle, pylint, and diff_quality.
+"""
+
+import re
+import subprocess
+import shlex
+import sys
+
+
+def fail_quality(name, message):
+    """
+    Fail the specified quality check.
+    """
+    print(name)
+    print(message)
+    sys.exit()
+
+
+def run_eslint():
+    """
+    Runs eslint on static asset directories.
+    If limit option is passed, fails build if more violations than the limit are found.
+    """
+    violations_limit = 1213
+
+    command = [
+        "node",
+        "--max_old_space_size=4096",
+        "node_modules/.bin/eslint",
+        "--ext", ".js",
+        "--ext", ".jsx",
+        "--format=compact",
+        "."
+    ]
+    print(shlex.join(command))
+    result = subprocess.run(
+        command,
+        text=True,
+        check=False,
+        capture_output=True
+    )
+
+    last_line = result.stdout.strip().splitlines()[-1] if result.stdout.strip().splitlines() else ""
+    regex = r'^\d+'
+    try:
+        num_violations = int(re.search(regex, last_line).group(0)) if last_line else 0
+    # An AttributeError will occur if the regex finds no matches.
+    except (AttributeError, ValueError):
+        print(f"FAILURE: Number of eslint violations could not be found in '{last_line}'")
```

Comment:
@salman2013 , simply printing an error message is this case is not sufficient. We need to be sure that the process fails (exits non-zero status code) so that the PR check fails. You can do so by adding `raise` to this block.